esp/services/ws_decoupledlogging/ws_decoupledloggingservice.cpp

```diff
@@ -155,53 +206,130 @@ bool CWSDecoupledLogEx::onGetLogAgentSetting(IEspContext& context, IEspGetLogAge
+                if (agentThread)
+                {
+                    agentThread->getLogRequestReader()->setPause(pause);
+                    agentStatus->setStatus(pause ? "Pausing" : "Resuming");
+                }
+                else
+                {
+                    agentStatus->setStatus("NotFound");
+                }
+
+                agentStatusResp.append(*agentStatus.getClear());
+            }
+            groupStatus->setLogAgentStatus(agentStatusResp);
+            groupStatusResp.append(*groupStatus.getClear());
         }
+        resp.setStatus(groupStatusResp);
+    }
+    catch(IException* e)
+    {
+        FORWARDEXCEPTION(context, e, ECLWATCH_INTERNAL_ERROR);
     }
+    return true;
+}
 
-    StringBuffer errorStatus;
-    checkLogAgentInList(agentNames, logAgentFound, errorStatus);
-    if (!errorStatus.isEmpty())
-        resp.setErrorStatus(errorStatus);
+void CWSDecoupledLogEx::pauseAllLoggingAgentsInGroup(WSDecoupledLogAgentGroup* group, bool pause, IArrayOf<IEspLogAgentStatus>& agentStatusResp)
+{
+    std::map<std::string, Owned<IUpdateLogThread>>&  agentThreadMap = group->getLoggingAgentThreads();
+    for (auto mt : agentThreadMap)
+    {
+        mt.second->getLogRequestReader()->setPause(pause);
 
-    return true;
+        Owned<IEspLogAgentStatus> agentStatus = createLogAgentStatus();
+        agentStatus->setAgentName(mt.first.c_str());
+        agentStatus->setStatus(pause ? "Pausing" : "Resuming");
+        agentStatusResp.append(*agentStatus.getClear());
+    }
 }
 
-void CWSDecoupledLogEx::checkLogAgentInList(StringArray& namesToCheck, BoolHash& namesFound, StringBuffer& status)
+void CWSDecoupledLogEx::getSettingsForAllLoggingAgentsInGroup(WSDecoupledLogAgentGroup* group, IArrayOf<IEspLogAgentSetting>& agentSettingResp)
 {
-    ForEachItemIn(i, namesToCheck)
+    std::map<std::string, Owned<IUpdateLogThread>>&  agentThreadMap = group->getLoggingAgentThreads();
+    for (auto mt : agentThreadMap)
     {
-        const char* name = namesToCheck.item(i);
-        bool* found = namesFound.getValue(name);
-        if (!found || !*found)
-            status.appendf("LogAgent %s not found. ", name);
+        CLogRequestReaderSettings* settings = mt.second->getLogRequestReader()->getSettings();
+
+        Owned<IEspLogAgentSetting> agentSetting = createLogAgentSetting();
+        agentSetting->setAgentName(mt.first.c_str());
+        agentSetting->setAgentStatus("Found");
+        agentSetting->setAckedFileList(settings->ackedFileList);
+        agentSetting->setAckedLogRequestFile(settings->ackedLogRequestFile);
+        agentSetting->setWaitSeconds(settings->waitSeconds);
+        agentSetting->setPendingLogBufferSize(settings->pendingLogBufferSize);
+        agentSettingResp.append(*agentSetting.getClear());
     }
 }
+
+IUpdateLogThread* WSDecoupledLogAgentGroup::getLoggingAgentThread(const char* name)
+{
+    auto match = loggingAgentThreads.find(name);
```

Comment:
Would a NULL check be advisable here, in case the method is called without checking? If I remember correctly, std::string is not fault tolerant WRT nullptr input.